### PR TITLE
Add lowering for aten.to.device

### PR DESCRIPTION
**Pull request by gpetters94**
As per #1075

### CODE REVIEW

**Review by pashu123 — APPROVED** [review]
LGTM.

**Comment by pashu123** [comment]
@gpetters94 Could you add an e2e test case?

**Review by vivekkhandelwal1 — COMMENTED** [review]
Is the lowering correct? Is it doing what the op is supposed to do?

**Comment by pashu123** [comment]
We can just fold the aten.to.device op. We don't have distributed/device semantics here. Pinging @ramiro050.

**Comment by gpetters94** [comment]
I'm not sure if I can add a test for this op. Its name should be `to.device` but it that's not valid Python syntax, so I'm not sure what I should be calling in the test.

**Comment by pashu123** [comment]
Please squash all your commits into one.

**Comment by ramiro050** [comment]
> I'm not sure if I can add a test for this op. Its name should be `to.device` but it that's not valid Python syntax, so I'm not sure what I should be calling in the test.

You need to specify enough arguments so that the right variant of `.to` gets used. If you do `torch.ops.aten.to(input, device='cpu', dtype=torch.float, non_blocking=False)`, you should get `to.device` generated

**Comment by ramiro050** [comment]
> We can just fold the aten.to.device op. We don't have distributed/device semantics here. Pinging @ramiro050.

This op is a bit strange, because it also has the ability of changing the `dtype` of the tensor, so we can't just fold it away. An option would be to use a canonicalization pattern, but the pattern would have to be conditional on the input being a value-semantics tensor (see https://github.com/llvm/torch-mlir/issues/1075#issuecomment-1187776230), so that it does not get applied too early. I personally think it is simpler to just do a decomposition

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -1996,6 +1996,23 @@ class DecomposeAtenToDtypeLayoutOp
 };
 } // namespace
 
+namespace {
+// Decompose `aten.to.device` op into `aten.to.dtype` op.
+class DecomposeAtenToDeviceOp : public OpRewritePattern<AtenToDeviceOp> {
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:2001]: Drive-by comment: I would add a comment here mentioning that once we get to decompositions, the `device` argument no longer means anything in torch-mlir, so we can drop that information.

**Comment by pashu123** [comment]
@gpetters94 Please rebase and push.
